plugins/org.eclipse.embedcdt.managedbuild.cross.arm.core/src/org/eclipse/embedcdt/managedbuild/cross/arm/core/EnvironmentVariableSupplier.java

```diff
@@ -139,34 +133,28 @@ public static PathEnvironmentVariable create(IConfiguration configuration) {
 				toolchainPath = deprecatedPersistentPreferences.getToolchainPath(toolchainId, toolchainName, project);
 			}
 
-			if (path.isEmpty()) {
-				path = toolchainPath;
-			} else {
-				if (!toolchainPath.isEmpty()) {
-					// Concatenate build tools path with toolchain path.
-					path += EclipseUtils.getPathSeparator();
-					path += toolchainPath;
-				}
+			if (!toolchainPath.isEmpty()) {
+				path.add(toolchainPath);
 			}
 
-			if (!path.isEmpty()) {
+			if (!path.isEmpty())
+
+			{
 				// if present, substitute macros
-				if (path.indexOf("${") >= 0) {
-					path = resolveMacros(path, configuration);
-				}
+				String pathVar = path.stream().map(p -> {
+					if (p.indexOf("${") >= 0) {
+						p = resolveMacros(p, configuration);
+					}
+					return p;
+				}).filter(Predicate.not(String::isBlank)).collect(Collectors.joining(File.pathSeparator));
 
-				File sysroot = new File(path);
-				// File bin = new File(sysroot, "bin"); //$NON-NLS-1$
-				// if (bin.isDirectory()) {
-				// sysroot = bin;
-				// }
 				if (DEBUG_PATH) {
 					if (Activator.getInstance().isDebugging()) {
-						System.out.println("arm.PathEnvironmentVariable.create() PATH=\"" + sysroot + "\" cfg="
+						System.out.println("arm.PathEnvironmentVariable.create() PATH=\"" + path + "\" cfg="
```

Comment:
Probably should be `pathVar`
```suggestion
						System.out.println("arm.PathEnvironmentVariable.create() PATH=\"" + pathVar + "\" cfg="
```